Fix subscription id removal

v-iam opened the pull request:
This PR fixes #16 and #23, to ensure that subscription ID is correctly removed in all cases that `SubscriptionRecordingProcessor` purports to address.

I recommend a new release (0.5.0, probably, since this changes the behavior) after this is merged. Failing to remove the subscription ID seems like a significant problem.

Commit:
Fix subscription ID removal and test new cases

Commit:
Fix no-self-use linter complaint

msftclas commented:
@v-iam,
Thanks for your contribution as a Microsoft full-time employee or intern. You do not need to sign a CLA.
_Thanks,
Microsoft Pull Request Bot_

troydai commented:
I'll test it first.

troydai committed:
Downgrade the version of vcrpy dependency

from 1.11.1 to 1.11.0 due to a sympton similar to this
https://github.com/kevin1024/vcrpy/issues/318

troydai commented:
Add a commit to this PR and it looks good to me. @v-iam you can merge and bump the version and release. I'll do some additional test on CLI side before we begin consuming new release.
